nodejs/helpers.js

```diff
@@ -60,18 +60,19 @@ module.exports = {
     /**
      * In the guide, https://docs.mongodb.com/ecosystem/use-cases/client-side-field-level-encryption-guide/,
      * we create the data key and then show that it is created by
-     * using a find_one query. Here, in implementation, we only create the key if
-     * it doesn't already exist, ensuring we only have one local data key.
+     * retreiving it using a findOne query. Here, in implementation, we only
+     * create the key if it doesn't already exist, ensuring we only have one
+     * local data key.
      */
     async findOrCreateDataKey() {
       const client = await new MongoClient(this.connectionString, {
```

Comment:
for the sake of education, I find it useful to separate these two things into "creating a client" and "connecting the client":

```
const client = new MongoClient('mongodb://localhost:27017', { options });
await client.connect();
```

In the future, there will be no `connect` so this degrades gracefully there as well

Reply:
You need to `await` the close of this client as well. Why not reuse the non-encrypted client for this task?

Reply:
I'm unsure what task you are referring to. This method creates a data key if one doesn't exist. Does this not require an encrypted client?